File: kidsWithCandies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        size=len(candies)
        flags=[False for _ in range(size)]
        maxCandies=max(candies)
        for i in range(size):
            if candies[i]+extraCandies>=maxCandies:
                flags[i]=True
        return flags
        

File: kidsWithCandies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        
        int maxCandies=0;
        for (int numCandies: candies) {
            if (numCandies>maxCandies) maxCandies=numCandies;
        }
        List<Boolean> result = new ArrayList<Boolean>();

        for (int numCandies: candies) {
            if (numCandies+extraCandies>=maxCandies) result.add(true);
            else result.add(false);
        }
        return result;
    }
}
